lib/src/common/disposable_state.dart

```diff
@@ -1,7 +1,16 @@
 /// Lifecycle states of a Disposable instance.
 enum DisposableState {
+  /// The object has been created.
   initialized,
+
+  /// The object has been instructed to dispose but it has not yet
+  /// begun the process because it is waiting on managed futures to
+  /// complete.
```

Comment:
I felt like this one should be documented and I went ahead and documented the others as well.